pkg/pillar/devicenetwork/devicenetwork.go

```diff
@@ -274,18 +276,90 @@ func devPortInstallAPname(ifname, ap string) {
 		}
 	}
 
-	if len(ap) == 0 {
+	if len(wconfig.APN) == 0 {
 		return
 	}
+
 	file, err := os.Create(filepath)
 	if err != nil {
 		log.Errorln(err)
 		return
 	}
 
-	log.Infof("devPortInstallAPname: write file %s for name %s", filepath, ap)
-	file.WriteString(ap)
+	for _, apn := range wconfig.APN {
+		s := fmt.Sprintf("%s\n", apn)
+		file.WriteString(s)
+		break // only handle the first APN for now utill we know how to handle multiple of APNs
+	}
 	file.Close()
+	log.Infof("devPortInstallAPname: write file %s for name %s", filepath, wconfig.APN)
+}
+
+func devPortInstallWifiConfig(ifname string, wconfig types.WirelessConfig) bool {
+	if _, err := os.Stat(wpaFilename); err == nil {
+		if err := os.Remove(wpaFilename); err != nil {
+			log.Errorln(err)
+			return false
+		}
+	}
+
+	file, err := os.Create(wpaFilename)
+	if err != nil {
+		log.Errorln(err)
+		return false
+	}
```

Comment:
I think it makes sense to follow the pattern of creating the new filename as a tmpfile (in the same directory) and then doing an atomic rename. That avoids having wpa_supplicant seeing the file as missing or partially written.

We have such code in pubsub.WriteRename and in UpdateSshAuthorizedKeys.

Reply:
ok. make sense. will check the example mentioned.